### DISCUSSION TITLE
Asenkron transactions

### DISCUSSION THREAD

**Discussion by SecondLifes** [viniciussanchez/RESTRequest4Delphi#182] [Ideas]
I think asynchronous operations are missing in this library. Are there any asynchronous functions among the plans?

**Comment by viniciussanchez** [comment]
![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/16382981/203ce0ed-e9b4-4e76-bce0-fa66dd213dab)

Currently there are these options... but if you need something more, you can send us a pull request.

**Reply by SecondLifes** [reply]
Bug
function TRequestClient.SynchronizedEvents(const AValue: Boolean): IRequest;
begin
  FRESTClient.SynchronizedEvents := AValue;
  FRESTRequest.SynchronizedEvents := AValue;
  -->Result := Self;
end;

And

**When you make changes, FBody gives an error.**


![2](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/4446775/db82724d-6a8e-4e92-9222-6da431165529)


![Untitled](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/4446775/1ea4526c-4dc0-47ce-a2ce-5f84e62e310a)

**Reply by matheusbarzon** [reply]
If I understood correctly, I believe this PR would solve the problem
https://github.com/viniciussanchez/RESTRequest4Delphi/pull/184

**Reply by SecondLifes** [reply]
Unfortunately, the body now comes empty. Error persists. I think you should run a test

**Reply by SecondLifes** [reply]
Thank you for your help, everything is fine.
